src/neuroconv/tools/roiextractors/imagingextractordatachunkiterator.py

```diff
@@ -114,35 +114,45 @@ def _get_scaled_buffer_shape(self, buffer_gb: float, chunk_shape: tuple) -> tupl
         assert buffer_gb > 0, f"buffer_gb ({buffer_gb}) must be greater than zero!"
         assert all(np.array(chunk_shape) > 0), f"Some dimensions of chunk_shape ({chunk_shape}) are less than zero!"
 
-        image_size = self._get_maxshape()[1:]
-        min_buffer_shape = tuple([chunk_shape[0]]) + image_size
+        series_max_shape = self._get_maxshape()[1:]
+        min_buffer_shape = tuple([chunk_shape[0]]) + series_max_shape
         scaling_factor = math.floor((buffer_gb * 1e9 / (math.prod(min_buffer_shape) * self._get_dtype().itemsize)))
-        max_buffer_shape = tuple([int(scaling_factor * min_buffer_shape[0])]) + image_size
+        max_buffer_shape = tuple([int(scaling_factor * min_buffer_shape[0])]) + series_max_shape
         scaled_buffer_shape = tuple(
             [
                 min(max(int(dimension_length), chunk_shape[dimension_index]), self._get_maxshape()[dimension_index])
                 for dimension_index, dimension_length in enumerate(max_buffer_shape)
             ]
         )
+
         return scaled_buffer_shape
 
     def _get_dtype(self) -> np.dtype:
         return self.imaging_extractor.get_dtype()
 
     def _get_maxshape(self) -> tuple:
-        video_shape = (self.imaging_extractor.get_num_frames(),)
-        image_shape = self.imaging_extractor.get_image_size()
-        width, height = image_shape[1], image_shape[0]  # ROIExtractors convention is flipped
-        video_shape += (width, height)
-        if len(image_shape) == 3:
-            depth = image_shape[2]
-            video_shape += (depth,)
-        return video_shape
+
```

Comment:
This is fixing a bug because now image_shape always returns the shape of the image while previously it could return the shape of the volume.

Once the next version of roiextractors is released we can make a better fix for this but at the moment the simplest and safest option is just to get a sample and look at the shape.